File: FMAplication/Domain/DailyActivities/DailyPOSM.cs
using FMAplication.Core;
using FMAplication.Domain.Reports;
using FMAplication.Enumerations;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace FMAplication.Domain.DailyActivities
{

    public class DailyPOSM : AuditableEntity<int>
    {
        public bool IsPOSMInstallation { get; set; }
        public bool IsPOSMRepair { get; set; }
        public bool IsPOSMRemoval { get; set; }
        public int DailyCMActivityId { get; set; }
        public int POSMInstallationStatus { get; set; }
        public int POSMRepairStatus { get; set; }
        public int POSMRemovalStatus { get; set; }
        [StringLength(512)]
        public string POSMRemovalIncompleteReason { get; set; }
        [StringLength(512)]
        public string POSMInstallationIncompleteReason { get; set; }
        [StringLength(512)]
        public string POSMRepairIncompleteReason { get; set; }
        public List<POSMReport> POSMProducts { get; set; } = new List<POSMReport>();   

    }
}

File: FMAplication/ClientApp/src/app/Shared/Entity/Daily-posm/daily-posm.ts
import { PosmInstallationStatus } from '../../Enums/posm-installation-status.enum';

export class DailyPosm {
    public id: number;
    public isPOSMInstallation: boolean;
    public isPOSMRepair: boolean;
    public isPOSMRemoval: boolean;
    public dailyCMActivityId: number;

    public posmInstallationStatus: string;
    public posmRepairStatus: string;
    public posmRemovalStatus: string;
    

    public  posmRemovalIncompleteReason: string;

    public  posmInstallationIncompleteReason: string;

    public posmRepairIncompleteReason : string;

    constructor() {

        this.posmInstallationStatus = Object.keys(PosmInstallationStatus).filter(k => isNaN(Number(k)))[0];
        this.posmRepairStatus = Object.keys(PosmInstallationStatus).filter(k => isNaN(Number(k)))[0];
        this.posmRemovalStatus = Object.keys(PosmInstallationStatus).filter(k => isNaN(Number(k)))[0];

    }

}
